httpdocs/ajax/proto/list.php

```diff
@@ -0,0 +1,55 @@
+<?php # This script list available protos
+  function error($message) {
+    die("{\"error\":\"$message\"}");
+  }
+  header('Content-Type: application/json');
+  $json = file_get_contents('php://input');
+  $data = json_decode($json);
+  require '../../../php/database.php';
+  $mysqli = new mysqli($database_host, $database_username, $database_password, $database_name);
+  if ($mysqli->connect_errno)
+    error("Can't connect to MySQL database: $mysqli->connect_error");
+  $mysqli->set_charset('utf8');
+  if (isset($data->url)) {
+    $url = $data->url;
+    $query = "UPDATE proto SET viewed = viewed + 1 WHERE url LIKE \"$url\"";
+    $mysqli->query($query) or error($mysqli->error);
+    die('{"status":"updated"}');
+  }
+  $sortBy = isset($data->sortBy) && $data->sortBy != "default" && $data->sortBy != "undefined" ?
+    $mysqli->escape_string($data->sortBy) : "viewed-desc";
+  $parameter = explode("-", $sortBy)[0];
+  $order = explode("-", $sortBy)[1];
+  if ($parameter == "title" || $parameter == "version") {
+    if ($order == "asc")
+      $order = "desc";
+    else
+      $order = "asc";
```

Comment:
We usually use capital letters for MySQL keywords:
```suggestion
      $order = "DESC";
    else
      $order = "ASC";
```